src/api/controllers/v1/nls/workflow.go

```diff
@@ -139,3 +140,40 @@ func (u WorkflowController) RerunWorkflow(c *gin.Context) {
 	}
 	c.Status(200)
 }
+
+// GetWorkflowByName
+//	@Summary	Get a workflow by name
+//	@Param		name	path	string	true	"name of workflow"
+//	@Tags		Workflow Management
+//	@Accept		json
+//	@Produce	json
+//	@Success	200	{object}	v1alpha1.Workflow
+//	@Failure	400	{object}	utils.ResponseError
+//	@Failure	404	{object}	utils.ResponseError
+//	@Failure	500	{object}	utils.ResponseError
+//	@Router		/nls/v1/workflows/{name} [get]
+func (u WorkflowController) GetWorkflowByName(c *gin.Context) {
+	wfName := c.Param("name")
+	
+	var workflow interface{}
+	var err error
+	retries := 3
+	delay := 15 * time.Second
```

Comment:
15 seconds seems like a long time to sleep between calls especially since this is tied back to a API call.   What do you think of shortening the sleep to 5 seconds and adding more retries?    If you have an architectural reason for the 15 seconds then disregard.

Reply:
I have updated the sleep to 5 seconds and retry count to 10